storm-client/src/jvm/org/apache/storm/daemon/worker/WorkerState.java

```diff
@@ -401,6 +418,7 @@ public void refreshConnections() {
             }
         }
 
+        suicideIfLocalAssignmentsChanged(assignment);
```

Comment:
We can place this as earlier as possible in this refreshConnections method.

Reply:
Addressed